Bottom: 3a44edcae1d59a9d89e7ed3bb7435d5b972c2479
Top:    919118ae1e1ec1be1c7d05214cb2b27d9b605ff0
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2015-01-04 22:02:11 +0000

Refresh of disable-drc-checks-on-outline-

---

diff --git a/src/find.c b/src/find.c
index c402568..15e40a1 100644
--- a/src/find.c
+++ b/src/find.c
@@ -728,11 +728,10 @@ LookupLOConnectionsToPVList (int flag, bool AndRats)
         {
           LayerType *layer = LAYER_PTR (layer_no);
 
-          if (layer->no_drc)
-             continue;
+          if (layer->no_drc || is_layer_mechanical (layer))
+            continue;
 
           info.layer = layer_no;
-
           /* add touching lines */
           if (setjmp (info.env) == 0)
             r_search (layer->line_tree, &search_box,
@@ -3635,6 +3634,9 @@ DRCAll (void)
     {
       COPPERLINE_LOOP (PCB->Data);
       {
+        if (is_layer_mechanical (layer))
+          continue;
+
         /* check line clearances in polygons */
         if (PlowsPolygon (PCB->Data, LINE_TYPE, layer, line, drc_callback, &info))
           {
@@ -3680,6 +3682,8 @@ DRCAll (void)
     {
       COPPERARC_LOOP (PCB->Data);
       {
+        if (is_layer_mechanical (layer))
+          continue;
         if (PlowsPolygon (PCB->Data, ARC_TYPE, layer, arc, drc_callback, &info))
           {
             IsBad = true;
diff --git a/src/misc.c b/src/misc.c
index 31021c9..d453c15 100644
--- a/src/misc.c
+++ b/src/misc.c
@@ -2419,6 +2419,14 @@ ElementOrientation (ElementType *e)
   return dy > 0 ? 3 : 1;
 }
 
+bool is_layer_mechanical (LayerType *layer)
+{
+  if (strcmp (layer->Name, "outline") == 0 ||
+      strcmp (layer->Name, "route") == 0)
+    return true;
+  return false;
+}
+
 int
 ActionListRotations(int argc, char **argv, Coord x, Coord y)
 {
diff --git a/src/misc.h b/src/misc.h
index d6f33e5..b0b254d 100644
--- a/src/misc.h
+++ b/src/misc.h
@@ -138,6 +138,8 @@ char * GetInfoString (void);
    comparing two similar footprints.  */
 int ElementOrientation (ElementType *e);
 
+bool is_layer_mechanical (LayerType *layer);
+
 /* These are in netlist.c */
 
 void NetlistChanged (int force_unfreeze);
